app/javascript/gobierto_data/webapp/components/Dataset.vue

```diff
@@ -782,9 +789,37 @@ export default {
         this.vizName = null
 
         await this.getPrivateVisualizations()
+
+        /* Check if the user saved a viz from another user, we need to wait to obtain the private visualizations to avoid error because it's possible which this Visualization is the first Visualization which user save */
+        if (user !== userId || newViz) {
+          await this.updateURL(newViz)
+        }
         await this.getPublicVisualizations()
       }
     },
+    updateURL(element) {
+      const {
+        name: nameComponent,
+        params: {
+          id: slugDataset
+        }
+      } = this.$route;
+
+      const { id: newId } = element
+      //Changes the path depending on if we save a query or viz.
+      const pathQueryOrViz = this.savingViz ? 'v' : 'q'
+
+      /*Don't updates the URL if the component is Editor, only replace and don't reload, because in the editor we've two options, saved a query or viz, if the user saves a viz, and we update the URL, the browser reloads, and the user goes to visualization tab, and this behavior is too hacky.*/
+      //https://developer.mozilla.org/en-US/docs/Web/API/History/pushState
+      history.pushState(
+        {},
+        null,
+        `${location.origin}/datos/${slugDataset}/${pathQueryOrViz}/${newId}`
+      )
```

Comment:
Nope, because `$router.push` and `$router.replace` reload the entire page, so if a user saves a visualization in the editor, we will redirect to the visualization tab